### DISCUSSION TITLE
ZodResolver is flattening the nested objects in the zod schema.

### DISCUSSION THREAD

**Discussion by KarthikRaju391** [colinhacks/zod#3294] [Q&A] [answered]
The only thing is that I am not manually defining the zod schema. Instead, I am generating a zod schema from a typescript object.

```tsx
const FormSchema = ZodSchemaGenerator(data, z.object({}))
```

When I am parsing it manually, it's working correctly. 
```ts
	console.log(
		FormSchema.parse({
			title: 'test',
			overview: {
				metabar: {
					metabarTitle: 'met title',
				},
				description: getEmptyEditorState(),
			},
		})
	)
```
but when inferring the types using `z.infer<typeof FormSchema>`

The result is as follows:
```ts
{
   description: undefined
   metabarTitle: undefined
   title: ""
}
```

It is not keeping the nested object fields like `overview` and `metabar`

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.

![image](https://github.com/colinhacks/zod/assets/31667350/3ed088d2-65da-4492-83ee-99b6093ab88d)
![image](https://github.com/colinhacks/zod/assets/31667350/33ea712a-5daa-45a0-b703-0dcbe0cf7391)

**Reply by KarthikRaju391** [reply]
Sorry! I forgot to close this discussion because the issue got solved. The problem was that I was not setting default values in case of objects.
